infor-design/enterprise
# 3951 / 4004 - Improve Documentation around Toolbar Flex/Contextual Action Panel and small CAP bugfix

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR Addresses a few issues:
 - The Contextual Action Panel didn't previously expose access to its inner Toolbar or Toolbar Flex API.  This PR adds a `toolbarAPI` property that returns access to the Toolbar.
- Per #3951, exposing the Toolbar API allows for access to the toolbar items and lets a developer enable/disable Toolbar buttons using these APIs.  Documentation for how to do this, along with some tests for related use cases have been added.
- There was no documentation page for Toolbar Flex and #4004 was requested to add good documentation on how to configure Flex Toolbars.  This documentation has been added and linked to the new CAP docs.
- There was a small bug where some examples of our jQuery-based CAP components were not opening anymore.  Added a typecheck on the CAP side for `data-modal` properties to fix.

**Related github/jira issue (required)**:
Closes #3951 (CAP API documentation)
Closes #4004 (Toolbar Flex API documentation)

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the demoapp, then:

_Read the docs:_
- Look through the new [Contextual Action Panel Toolbar API documentation](https://github.com/infor-design/enterprise/compare/3951-document-cap-button-api?expand=1#diff-90c572f788548d96785d37a4d7662970R62-R74).
- Look through the new [Flex Toolbar API documentation](https://github.com/infor-design/enterprise/compare/3951-document-cap-button-api?expand=1#diff-2656e542ed59d2af601e13be0192495bR1).
- See if all of it makes sense / has errors

_Test disabling a CAP Toolbar button:_
- Open http://localhost:4000/components/contextualactionpanel/test-disable-button.html
- Click the button labeled "Open the CAP Flex Toolbar API Test".  The CAP should open (demos the small bugfix)
- When the CAP opens, click the button labeled "Disable the Close Button"
- The Close button in the Flex Toolbar above should have been disabled. (demos the API exposure)

This is also covered in new functional and e2e tests.

**Included in this Pull Request**:
